### DISCUSSION TITLE
See what a custom command or alias does

### DISCUSSION THREAD

**Discussion by BatmanAoD** [nushell/nushell#10743] [Q&A]
`which` will tell you whether a command is an alias or custom command, but will not tell you what the definition is.

You can see the definition of an alias using `scope aliases | where name == <name>`, but is there a simpler way?

As far as I can tell, there is no way to see the definition of a custom command.

For comparison, in `sh`-style shells, `type <foo>` gives you all of this information:

* What kind of command `foo` is
* For external commands, what the path is (and even whether it has been hashed!)
* For aliases, the expansion
* For functions, the function definition

For the most part, Nushell's `which` behaves more like the traditional `type` builtin than it does like the traditional `which` utility. But since it doesn't expand aliases or custom commands, it's a bit more limited, which is unfortunate.

**Comment by fdncred** [comment]
There's also `help aliases` but it's identical to `scope aliases`. I thought `view source` would work but it doesn't.
